templates/config.json.j2

```diff
@@ -91,9 +91,14 @@
     {% endif %}
     {% if consul_tls_enable | bool %}
     "ca_file": "{{ consul_tls_dir }}/{{ consul_tls_ca_crt | basename }}",
+    {% if auto_encrypt is not defined or (auto_encrypt is defined and not auto_encrypt.enabled | bool)
+        or (item.config_version != 'client') | bool %}
```

Comment:
Hi @paladin-devops! Nice work, I was looking for a fix for this :)

I have a suggestion to simplify this conditional — the same would apply to the second `when` condition in `tasks/tls.yml`.

```suggestion
    {% if (item.config_version != 'client') or not (auto_encrypt.enabled | default(false) | bool) %}
```

- I put `(item.config_version != 'client')` first to make it more explicit that it doesn't apply to server nodes.
- `(item.config_version != 'client')` is a comparison expression, so casting to `bool` should be unnecessary.
- I rewrote the `auto_encrypt.enabled` part in fewer words, it seems to me easier to understand.